components/liked.js

```diff
@@ -0,0 +1,65 @@
+import { useReducer, useCallback } from "react";
+
+import {
+  HeartIcon,
+} from "@heroicons/react/solid";
+import {
+  HeartIcon as HeartOutlineIcon,
+} from "@heroicons/react/outline";
+
+const likeReducer = (state, action) => {
+  switch (action.type) {
+    case "loading":
+      return {
+        isLoading: true,
+        liked: action.liked,
+      };
+    case "error":
+      return {
+        ...state,
+        isLoading: false,
+        liked: previousValue.liked,
+        error: action.error,
+      };
+    case "success":
+      return {
+        ...state,
+        isLoading: false,
+        previousValue: action.liked,
+        liked: action.liked,
+      };
+    default: {
+      throw new Error(`Unhandled action type: ${action.type}`);
+    }
+  }
+}
+
+const useLikeState = (initialLiked) => {
+  const [state, dispatch] = useReducer(likeReducer, {
+    isLoading: false,
+    liked: initialLiked,
+    previousValue: initialLiked,
+    error: null,
+  });
+
+  // TODO: TD082321 - this should be debounced
```

Comment:
Good thing to call out! But also we dont' need it now..

Reply:
Right, that's why it's a todo